Backend/Remora.Discord.Core/Optional.cs

```diff
@@ -60,25 +60,34 @@ namespace Remora.Discord.Core
             }
         }
 
+        // CS8775: Member must have a non-null value when exiting in some condition.
         // Technically, this is a lie, but since the nullability of the type is reliant on the actual generic type and
         // its annotations, this warning can be disabled.
         #pragma warning disable CS8775
         /// <inheritdoc />
+        [Obsolete("Prefer Optional{TValue}.IsSpecified")]
         [MemberNotNullWhen(true, nameof(Value))]
```

Comment:
This attribute, and the corresponding warning disabling above, does have room for improvement. I realize this isn't an optimal way of doing things, but it produces correct results in the majority of cases (e.g, when T is not a T?, be it struct or class). It is an attempt to convince the nullability annotations to warn correctly for the most common usages; I fully admit it's not always correct, though.

My preference would be to remove this attribute and figure out a better way of producing the correct nullability warnings depending on the contained type.